chain/client/src/client_actor.rs

```diff
@@ -1941,27 +1961,33 @@ impl Handler<ApplyStatePartsResponse> for ClientActor {
     }
 }
 
-impl Handler<BlockCatchUpRequest> for SyncJobsActor {
+impl Handler<WithSpanContext<BlockCatchUpRequest>> for SyncJobsActor {
     type Result = ();
 
-    fn handle(&mut self, msg: BlockCatchUpRequest, _: &mut Self::Context) -> Self::Result {
-        let _span =
-            tracing::debug_span!(target: "client", "handle", handler = "BlockCatchUpRequest")
-                .entered();
+    fn handle(
+        &mut self,
+        msg: WithSpanContext<BlockCatchUpRequest>,
+        _: &mut Self::Context,
+    ) -> Self::Result {
+        let (_span, msg) = handler_span!("client", "SyncJobsActor", "BlockCatchUpRequest", msg);
```

Comment:
Just realizing, but instead of specifying the string-ified name of `Self` everywhere, you could probably have the compiler figure it out for you. See `self_name` in [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3fd65cc190dabdcbfd34a0bc22f6365e).

You can do the same with the type of message too. Using something along the lines of:

```
fn type_name_of<T>(_: &T) {
    let name = std::any::type_name::<T>();
    name.rsplit_once("::").map_or(name, |(_, name)| name)
}

type_name_of(&msg.msg)
```

with this, `handler_span` invocations can become a simple and fairly fool-proof

```
let (_span, msg) = handler_span!("client", msg); 
```

(and if additional flexibility _is_ necessary, additional macro matchers can be added to allow overriding the default behaviour for the actor and message names.)

Reply:
This works surprisingly well, thank you.